### PR TITLE
Pathing Issue

### PR DESCRIPTION
**Pull request by KyroVibe**
### Description
- Changed file field in parts to store file name
- Added field to convert file name to absolute path
- Replaced usage of file name with new absolute path

### CODE REVIEW

**Review by BrandonPacewic — COMMENTED** [review]
Am I correct in saying that the robot builder parts will need to be created again and updated in the installer? Because I used the installer to get the parts in the first place it still will not work even with the current changes.

**Comment by KyroVibe** [comment]
> Am I correct in saying that the robot builder parts will need to be created again and updated in the installer? Because I used the installer to get the parts in the first place it still will not work even with the current changes.

Yes. I fixed the ones on my machine and remaking them should result in the corrected path

**Review by BrandonPacewic — COMMENTED** [review]
Seems to work as far as I can tell. Although I'll admit I may not have tested this correctly as I am on my personal computer.

File: engine/Assets/Scripts/SimObjects/MixAndMatch/MixAndMatchSaveData.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@ namespace SimObjects.MixAndMatch {
     public static class MixAndMatchSaveUtil {
         private static readonly char ALT_SEP = Path.AltDirectorySeparatorChar;
 
-        private static readonly string PART_MIRABUF_FOLDER_PATH =
+        public static readonly string PART_MIRABUF_FOLDER_PATH =
```
[the remark this answers is not in the document]
Comment (KyroVibe) [reply engine/Assets/Scripts/SimObjects/MixAndMatch/MixAndMatchSaveData.cs:12]: It's (basically) a constant for the file path to the folder and its used externally from the class

[the remark this answers is not in the document]
Comment (BrandonPacewic) [reply engine/Assets/Scripts/SimObjects/MixAndMatch/MixAndMatchSaveData.cs:12]: Gotcha.

**Review by BrandonPacewic — APPROVED** [review]
Seems ok. Send me a discord msg when you update the installer and have a new build. Want to test it again on my windows computer before we send it out.

**Review by LucaHaverty — CHANGES_REQUESTED** [review]
The fix works well when I create a new part, but it throws an error when loading an old part. Do you think you could add a check that fixes old parts in case people have them in appdata already when they install the update with this fix?

**Comment by KyroVibe** [comment]
@LucaHaverty I added a check to see if the new path is found, if not, look for the old path, then if that's not found, throw an exception. We really need to add better error handling to everything mira

**Review by LucaHaverty — APPROVED** [review]
Works well, great work Hunter!

**Pull request merged**